.github/workflows/main.yml

```diff
@@ -0,0 +1,93 @@
+name: CI
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up workspace
+        uses: ./.github/actions/setup
+
+      - name: Run linter
+        run: npm run lint
+
+  test-node:
+    name: Test on Node.js ${{ matrix.node-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up workspace
+        uses: ./.github/actions/setup
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Run tests
+        run: npm run test-node
+
+  test-browser:
+    name: Test in browser
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up workspace
+        uses: ./.github/actions/setup
+
+      - name: Install browsers
+        run: npx playwright install --with-deps
+
+      - name: Run tests
+        run: npm run test-browser
+
+  benchmark:
+    name: Benchmark
+    runs-on: ubuntu-latest
+    permissions:
+      # To deploy GitHub pages website.
+      deployments: write
+      # To update benchmark contents in gh-pages branch.
+      contents: write
```

Comment:
Is this actually deploying to Github pages (which comes form the root of the `main` branch)? I'm trying to track down evidence of this but don't see any.

Reply:
No, it's not. I intended to set up GitHub pates, but ended up removing it and forgot to clear the permissions here. I pushed a new change to remove the permissions.